HLTrigger/special/plugins/HLTScoutingRecHitProducer.cc

```diff
@@ -24,91 +26,145 @@ class HLTScoutingRecHitProducer : public edm::global::EDProducer<> {
 
 private:
   void produce(edm::StreamID, edm::Event& iEvent, edm::EventSetup const&) const final;
+  static void produceEcal(edm::Event& iEvent,
+                          const reco::PFRecHitCollection& inputRecHits,
+                          double minEnergyEB,
+                          double minEnergyEE,
+                          int mantrissaPrecision,
+                          const std::string tag = "");
+  static void produceHcal(edm::Event& iEvent,
+                          const reco::PFRecHitCollection& inputRecHits,
+                          double minEnergyHBHE,
+                          int mantrissaPrecision,
+                          const std::string tag = "");
+
+  template <typename T>
+  void setToken(edm::EDGetTokenT<T>& token, const edm::ParameterSet& iConfig, std::string name) {
+    const auto inputTag = iConfig.getParameter<edm::InputTag>(name);
+    if (!inputTag.encode().empty()) {
+      token = consumes(iConfig.getParameter<edm::InputTag>(name));
+    }
+  }
```

Comment:
One suggestion (but I haven't tested it): if we change a bit the structure of this function, I think we could keep the class data members `const`, and I think this usually helps making it obvious that the plugin can be `edm::global`.
```suggestion
  template <typename T>
  edm::EDGetTokenT<T> getToken(const edm::ParameterSet& iConfig, const std::string& name) const {
    edm::EDGetTokenT<T> token{};
    const auto inputTag = iConfig.getParameter<edm::InputTag>(name);
    if (!inputTag.encode().empty()) {
      token = consumes(inputTag);
    }
    return token;
  }
```

Reply:
sure its pure lazyness in that I did it that way as I dont have to specify the template type (although I didnt check if its now smart enough to figure it out by the token return type).